### PR TITLE
fix: apply layered z-indices across components

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #4372 #3099 #4682 #3724 #4781 

## Summary

This fixes a couple of layering issues we had in our components by doing the following:

1. updated tailwind config to introduce contextual z-index layers
2. updated component styles to reference these
3. removed unnecessary z-indices (see 3 below)

### Notes

1. I can break any of these items to separate PRs if deemed appropriate
2. Layers are based on https://www.duetds.com/tokens/#z-index and can be tweaked accordingly
3. Any removed `z-index` can be restored if we spot a regression. Removal was determined by:
    * evaluating the need for a stacking context
    * local testing
4. Repro cases from the referenced issues were tested and work as expected
5. Updated all z-indices to be applied as tailwind utils
6. Skipped adding CSS props for now until the z-index tokens are decided

### CODE REVIEW

**Comment by macandcheese** [comment]
Awesome! @SkyeSeitz @ashetland these layers will be useful to replicate in Figma / tokens / documentation

File: tailwind.config.ts [hunk 1]
```diff
@@ -208,6 +208,17 @@ export default {
       },
       maxHeight: {
         menu: "45vh"
+      },
+      zIndex: {
```
Comment (driskull) [inline tailwind.config.ts:212]: this is nice :)

File: src/components/shell-panel/shell-panel.scss [hunk 2]
```diff
@@ -35,7 +35,6 @@
   absolute
   bottom-0
   top-0
-  z-10
```
Comment (driskull) [inline src/components/shell-panel/shell-panel.scss:38]: Do we know why some of these are here? Just don't want to cause any regressions. Maybe more testing with the ones we remove?

File: src/components/panel/panel.scss [hunk 3]
```diff
@@ -67,12 +67,6 @@
   @apply hidden;
 }
 
-:host([loading]) .container,
```
Comment (driskull) [inline src/components/panel/panel.scss:70]: Did you test this to make sure that the scrim is showing correctly and with scrolling elements below?

File: src/components/action/action.scss [hunk 4]
```diff
@@ -199,7 +199,6 @@
   .button::after {
     content: "";
     @apply absolute
-      z-10
```
Comment (driskull) [inline src/components/action/action.scss:202]: Not sure why this was here 🤷

**Review by driskull — APPROVED** [review]
Great idea 💡

File: src/components/panel/panel.scss (hunk 3 above)
Comment (jcfranco) [reply src/components/panel/panel.scss:70]: Yes, but will test again to make sure.

File: src/components/shell-panel/shell-panel.scss (hunk 2 above)
Comment (jcfranco) [reply src/components/shell-panel/shell-panel.scss:38]: I couldn't deduce the need for it, and didn't notice any visual issues after removing. 

> Maybe more testing with the ones we remove?

Great idea.

File: src/components/panel/panel.scss (hunk 3 above)
Comment (jcfranco) [reply src/components/panel/panel.scss:70]: Local non-exhaustive testing looks good. 🎉

**Pull request merged**